package.json

```diff
@@ -34,17 +34,28 @@
 	"scripts": {
 		"build": "yarn clean && lerna run build --stream",
 		"clean": "lerna run clean",
-		"ganache": "npx ganache -m 'smart guide what forget tired jungle always expire rescue boring glue champion' --account=\"0x4c3758228f536f7a210f8936182fb5b728046970b8e3215d0b5cb4c4faae8a4e,100000000000000000000\" --account=\"0x34aeb1f338c17e6b440c189655c89fcef148893a24a7f15c0cb666d9cf5eacb3,100000000000000000000\"",
-		"ganache-background": "yarn ganache > /dev/null & client=$! npx wait-port 8545",
-		"geth": "docker run -p 8545:8545 ethereum/client-go --nodiscover --nousb --ws --ws.addr 0.0.0.0 --ws.port 8545 --http --http.addr 0.0.0.0 --http.port 8545 --allow-insecure-unlock --http.api personal,web3,eth,admin,debug,txpool,net --ws.api personal,web3,eth,admin,debug,miner,txpool,net --dev",
-		"geth-background": "docker run -d -p 8545:8545 ethereum/client-go --nodiscover --nousb --ws --ws.addr 0.0.0.0 --ws.port 8545 --http --http.addr 0.0.0.0 --http.port 8545 --allow-insecure-unlock --http.api personal,web3,eth,admin,debug,txpool,net --ws.api personal,web3,eth,admin,debug,miner,txpool,net --dev",
+		"ganache:start": "WEB3_SYSTEM_TEST_BACKEND=ganache && ./scripts/ganache.sh start",
+		"ganache:start:background": "WEB3_SYSTEM_TEST_BACKEND=ganache && ./scripts/ganache.sh start 1",
+		"ganache:stop": "./scripts/ganache.sh stop",
+		"geth:start": "WEB3_SYSTEM_TEST_BACKEND=geth && ./scripts/geth.sh start",
+		"geth:start:background": "WEB3_SYSTEM_TEST_BACKEND=geth && ./scripts/geth.sh start 1",
+		"geth:stop": "./scripts/geth.sh stop",
 		"lint": "lerna run lint --stream --parallel",
 		"lint:fix": "lerna run lint:fix --stream --parallel",
 		"format": "lerna run format --stream --parallel",
 		"test": "lerna run test --stream --parallel",
 		"test:unit": "lerna run test:unit --stream --parallel",
 		"test:integration": "lerna run test:integration --stream",
-		"test:integration:browsers": "./node_modules/.bin/karma start karma.config.js",
+		"test:e2e:electron": "lerna run test:e2e:electron --stream",
+		"test:e2e:chrome": "lerna run test:e2e:chrome --stream",
+		"test:e2e:firefox": "lerna run test:e2e:firefox --stream",
+		"test:e2e:ganache:http": "./scripts/test-runner.sh ganache http",
+		"test:e2e:ganache:ws": "./scripts/test-runner.sh ganache ws",
+		"test:e2e:geth:http": "./scripts/test-runner.sh geth http",
+		"test:e2e:geth:ws": "./scripts/test-runner.sh geth ws",
+		"test:e2e:browsers:electron": "./scripts/test-runner.sh ganache ws electron",
+		"test:e2e:browsers:chrome": "./scripts/test-runner.sh ganache ws chrome",
+		"test:e2e:browsers:firefox": "./scripts/test-runner.sh ganache ws firefox",
```

Comment:
What the difference between these tests from:

```json
"test:e2e:electron": "lerna run test:e2e:electron --stream",
"test:e2e:chrome": "lerna run test:e2e:chrome --stream",
"test:e2e:firefox": "lerna run test:e2e:firefox --stream",
```

Shouldn't they be called

```json
"test:e2e:electron:ws"
"test:e2e:chrome:ws"
"test:e2e:firefox:ws"
```

and shouldn't there be the `http` variants?

Reply:
Yes we can rename these tasks. The earlier pipeline was configured only for the `ws` on the browsers, so I used it same. I think as we already run the tests separately in integration for both `http` and `ws`, for browsers just the `ws` would be good enough.